### PR TITLE
Contribution guide to document actual guide for pull requests

### PR DESCRIPTION
**Pull request by khalidmammadov**
Currently contribution guide does not reflect actual flow to raise a new PR and hence it's not clear (for a new contributors) what exactly needs to be done to make a PR for Spark repository and test it as per expectation. This PR addresses that by following:

- It describes in the Pull request section of the Contributing page the actual procedure and takes a contributor through a step by step process.
- It removes optional "Running tests in your forked repository" section on Developer Tools page which is obsolete now and doesn't reflect reality anymore i.e. it says we can test by clicking “Run workflow” button which is not available anymore as workflow does not use "workflow_dispatch" event trigger anymore and was removed in  https://github.com/apache/spark/pull/32092 
- Instead it documents the new procedure that above PR introduced i.e. contributors needs to use their own GitHub free workflow credits to test new changes they are purposing and a Spark Actions workflow will expect that to be completed before marking PR to be ready for a review.  
- Some general wording was copied from  "Running tests in your forked repository" section on Developer Tools page but main content was rewritten to meet objective
- Also fixed URL to developer-tools.html to be resolved by parser (that converted it into relative URI) instead of using hard coded absolute URL. 

Tested imperically with `bundle exec jekyll serve` and static files were generated with `bundle exec jekyll build` commands

This closes https://issues.apache.org/jira/browse/SPARK-37996

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -321,11 +321,17 @@ Example: `Fix typos in Foo scaladoc`
...
+our GitHub Actions workflows automatically run tests for your pull request/following commits 
+and every run burdens the limited resources of GitHub Actions in Apache Spark repository.
+Below steps will take your through the process.
+
 
 1. <a href="https://help.github.com/articles/fork-a-repo/">Fork</a> the GitHub repository at 
 <a href="https://github.com/apache/spark">https://github.com/apache/spark</a> if you haven't already
-1. Clone your fork, create a new branch, push commits to the branch.
-1. Consider whether documentation or tests need to be added or updated as part of the change, 
+2. Go to "Actions" tab on your forked repository and enable "Build and test" and "Report test results" workflows
```
Comment (srowen) [inline contributing.md]: Just write `1.` on each of these and revert subsequent changes. The point is that markdown generates the correct numbering

File: contributing.md [hunk 2]
```diff
@@ -321,11 +321,17 @@ Example: `Fix typos in Foo scaladoc`
 `dev@spark.apache.org` first before proceeding to implement the change.
 
 <h3>Pull request</h3>
+Before creating a pull request in Apache Spark, it is important to check if tests can pass on your branch because
```
Comment (srowen) [inline contributing.md:325]: Newline

File: contributing.md [hunk 3]
```diff
@@ -354,46 +360,48 @@ and add them as needed.
...
 though you are welcome to comment that you have begun work.
-1. The Jenkins automatic pull request builder will test your changes
-     1. If it is your first contribution, Jenkins will wait for confirmation before building 
-     your code and post "Can one of the admins verify this patch?"
-     1. A committer can authorize testing with a comment like "ok to test"
-     1. A committer can automatically allow future pull requests from a contributor to be 
-     tested with a comment like "Jenkins, add to whitelist"
-1. After about 2 hours, Jenkins will post the results of the test to the pull request, along 
+10. The Jenkins automatic pull request builder will test your changes
+      1. If it is your first contribution, Jenkins will wait for confirmation before building
```
Comment (srowen) [inline contributing.md]: Revert these indent changes

File: developer-tools.md [hunk 4]
```diff
@@ -246,23 +246,6 @@ Kubernetes, and more importantly, minikube have rapid release cycles, and point
 
 Apache Spark leverages GitHub Actions that enables continuous integration and a wide range of automation. Apache Spark repository provides several GitHub Actions workflows for developers to run before creating a pull request.
 
-<a name="github-workflow-tests"></a>
```
Comment (srowen) [inline developer-tools.md:249]: Actually, I wonder if we can leave this here and point to it in the contributing guide?

**Review by srowen — CHANGES_REQUESTED** [review]

File: contributing.md (hunk 1 above)
Comment (dongjoon-hyun) [reply contributing.md]: +1 for Sean's comment.

File: contributing.md (hunk 2 above)
Comment (khalidmammadov) [reply contributing.md:325]: Done

File: developer-tools.md (hunk 4 above)
Comment (khalidmammadov) [reply developer-tools.md:249]: This process become obsolete after https://github.com/apache/spark/pull/32092 and is incorrect as it stands and hence this PR merges the new process to the Pull request section so it's clear for the new contributors.

File: contributing.md (hunk 3 above)
Comment (khalidmammadov) [reply contributing.md]: done

File: contributing.md (hunk 1 above)
Comment (khalidmammadov) [reply contributing.md]: Fixed

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — COMMENTED** [review]
This looks like mostly straight-forward moving the content from `developer-tools` page to `contributing` page. Could you make it clear in the PR title and description about what was copied and what was newly revised and added, @khalidmammadov ?

If you don't mind, please update the corresponding JIRA too.

**Comment by srowen** [comment]
A little more detail in the description would be nice; we don't need a JIRA for this website change I think

**Comment by dongjoon-hyun** [comment]
> A little more detail in the description would be nice; we don't need a JIRA for this website change I think

To @srowen: Yes, I agree with you that we don't need a JIRA for this website change. However, @khalidmammadov already filed SPARK-37996 on January 24th as he described in the AS-IS PR description. That's the reason why I asked him to update it.

**Review by HyukjinKwon — APPROVED** [review]
Yeah, updating PR / JIRA descriptions would be great. Otherwise, it looks fine to me 2/

**Comment by khalidmammadov** [comment]
I think all done, anything left to do here?

**Review by srowen — APPROVED** [review]

**Pull request closed without merging**